### ISSUE TITLE
Support mesh in the data catalogs 

### ISSUE THREAD

**Issue by alimeshgi** [Deltares/hydromt#417]
### Kind of request

Adding new functionality

### Enhancement Description

adding new data adapter for mesh

**Comment by DirkEilander** [comment]
@hboisgon it would be good to discuss if we want this in Q4. If so can you provide more context to this issue / refine it, potentially together with someone who good actually do the work?

**Comment by hboisgon** [comment]
I refined and added use cases to the issue. In terms of Q4 or Q1 I am not sure. @veenstrajelmer or @xldeltares, any feedback?

**Comment by veenstrajelmer** [comment]
Good idea to include ugrid support via xugrid. However, it would be nice to align it with moving dfm_tools modelbuilder features to hydromt_delft3dfm. If that is done, I think this is quite ambitious for Q4.

I was also triggered by the CMIP6 models that are not on a regulargrid. I have encountered CMCC on a curvilinear global/period grid, which is quite inconvenient but it can be sort of read as a ugrid. I have created this function in dfm_tools which might be a good start:
https://github.com/Deltares/dfm_tools/blob/29f9488fe9aba5993ea588fe3979bbc8eb6557b2/dfm_tools/xugrid_helpers.py#L236
Example script: https://github.com/Deltares/dfm_tools/blob/main/tests/examples_workinprogress/workinprogress_CMCC_plotting.py
However, I think there are more applications that benefit from this, so we might want to include it in xugrid or another curvi package instead of hydromt.

**Comment by hboisgon** [comment]
@veenstrajelmer thanks for your input. Let's move this to Q1 next year then. I'd be interested to have a chat to see how we can best combine hydromt, xugrid and what you did in dfm_tools
